src/pytest_bdd/gherkin_parser.py

```diff
@@ -0,0 +1,345 @@
+        )
+
+
+@dataclass
+class Rule:
+    id: str
+    keyword: str
+    location: Location
+    name: str
+    description: str
+    tags: list[Tag]
+    children: list[Child]
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            id=data["id"],
+            keyword=data["keyword"],
+            location=Location.from_dict(data["location"]),
+            name=data["name"],
+            description=data["description"],
+            tags=[Tag.from_dict(tag) for tag in data["tags"]],
+            children=[Child.from_dict(child) for child in data["children"]],
+        )
+
+
+@dataclass
+class Background:
+    id: str
+    keyword: str
+    location: Location
+    name: str
+    description: str
+    steps: list[Step]
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            id=data["id"],
+            keyword=data["keyword"],
+            location=Location.from_dict(data["location"]),
+            name=data["name"],
+            description=data["description"],
+            steps=[Step.from_dict(step) for step in data["steps"]],
+        )
+
+
+@dataclass
+class Child:
+    background: Background | None = None
+    rule: Rule | None = None
+    scenario: Scenario | None = None
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            background=Background.from_dict(data["background"]) if data.get("background") else None,
+            rule=Rule.from_dict(data["rule"]) if data.get("rule") else None,
+            scenario=Scenario.from_dict(data["scenario"]) if data.get("scenario") else None,
+        )
+
+
+@dataclass
+class Feature:
+    keyword: str
+    location: Location
+    tags: list[Tag]
+    name: str
+    description: str
+    children: list[Child]
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            keyword=data["keyword"],
+            location=Location.from_dict(data["location"]),
+            tags=[Tag.from_dict(tag) for tag in data["tags"]],
+            name=data["name"],
+            description=data["description"],
+            children=[Child.from_dict(child) for child in data["children"]],
+        )
+
+
+@dataclass
+class GherkinDocument:
+    feature: Feature
+    comments: list[Comment]
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            feature=Feature.from_dict(data["feature"]),
+            comments=[Comment.from_dict(comment) for comment in data["comments"]],
+        )
+
+
+def _to_raw_string(normal_string: str) -> str:
+    return normal_string.replace("\\", "\\\\")
+
+
+def get_gherkin_document(abs_filename: str, encoding: str = "utf-8") -> GherkinDocument:
+    with open(abs_filename, encoding=encoding) as f:
+        feature_file_text = f.read()
+
+    try:
+        gherkin_data = Parser().parse(TokenScanner(feature_file_text))
```

Comment:
do we need to use TokenScanner? from the gherkin README.md, it seems we can just do 
```
parser = Parser()
gherkin_document = parser.parse("Feature: ...")
```

Reply:
Good point.  Updated :)